src/test/storage/table_test.cpp

```diff
@@ -12,63 +12,71 @@
-
-// TEST_F(StorageTableTest, RowCount) {
-//   EXPECT_EQ(t.row_count(), 0u);
-//   t.append({4, "Hello,"});
-//   t.append({6, "world"});
-//   t.append({3, "!"});
-//   EXPECT_EQ(t.row_count(), 3u);
-// }
-
-// TEST_F(StorageTableTest, GetColumnName) {
-//   EXPECT_EQ(t.column_name(ColumnID{0}), "col_1");
-//   EXPECT_EQ(t.column_name(ColumnID{1}), "col_2");
-//   // TODO(anyone): Do we want checks here?
-//   // EXPECT_THROW(t.column_name(ColumnID{2}), std::exception);
-// }
-
-// TEST_F(StorageTableTest, GetColumnType) {
-//   EXPECT_EQ(t.column_type(ColumnID{0}), "int");
-//   EXPECT_EQ(t.column_type(ColumnID{1}), "string");
-//   // TODO(anyone): Do we want checks here?
-//   // EXPECT_THROW(t.column_type(ColumnID{2}), std::exception);
-// }
-
-// TEST_F(StorageTableTest, GetColumnIdByName) {
-//   EXPECT_EQ(t.column_id_by_name("col_2"), 1u);
-//   EXPECT_THROW(t.column_id_by_name("no_column_name"), std::exception);
-// }
-
-// TEST_F(StorageTableTest, GetChunkSize) { EXPECT_EQ(t.max_chunk_size(), 2u); }
+class StorageTableTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    t.add_column("col_1", "int");
+    t.add_column("col_2", "string");
+  }
+
+  Table t{2};
+};
+
+TEST_F(StorageTableTest, ChunkCount) {
+  EXPECT_EQ(t.chunk_count(), 1u);
+  t.append({4, "Hello,"});
+  t.append({6, "world"});
+  t.append({3, "!"});
+  EXPECT_EQ(t.chunk_count(), 2u);
+}
+
+TEST_F(StorageTableTest, GetChunk) {
+  t.get_chunk(ChunkID{0});
+  // TODO(anyone): Do we want checks here?
+  // EXPECT_THROW(t.get_chunk(ChunkID{q}), std::exception);
+  t.append({4, "Hello,"});
+  t.append({6, "world"});
+  t.append({3, "!"});
+  t.get_chunk(ChunkID{1});
+}
+
+TEST_F(StorageTableTest, ColumnCount) { EXPECT_EQ(t.column_count(), 2u); }
+
+TEST_F(StorageTableTest, RowCount) {
+  EXPECT_EQ(t.row_count(), 0u);
+  t.append({4, "Hello,"});
+  t.append({6, "world"});
+  t.append({3, "!"});
+  EXPECT_EQ(t.row_count(), 3u);
+}
+
+TEST_F(StorageTableTest, GetColumnName) {
+  EXPECT_EQ(t.column_name(ColumnID{0}), "col_1");
+  EXPECT_EQ(t.column_name(ColumnID{1}), "col_2");
+  // TODO(anyone): Do we want checks here?
+  // EXPECT_THROW(t.column_name(ColumnID{2}), std::exception);
+}
+
+TEST_F(StorageTableTest, GetColumnNames) { EXPECT_EQ(t.column_names(), (std::vector<std::string>{"col_1", "col_2"})); }
+
+TEST_F(StorageTableTest, GetColumnType) {
+  EXPECT_EQ(t.column_type(ColumnID{0}), "int");
+  EXPECT_EQ(t.column_type(ColumnID{1}), "string");
+  // TODO(anyone): Do we want checks here?
+  // EXPECT_THROW(t.column_type(ColumnID{2}), std::exception);
+}
+
+TEST_F(StorageTableTest, GetColumnIdByName) {
+  EXPECT_EQ(t.column_id_by_name("col_2"), 1u);
+  EXPECT_THROW(t.column_id_by_name("no_column_name"), std::exception);
+}
+
+TEST_F(StorageTableTest, GetChunkSize) { EXPECT_EQ(t.max_chunk_size(), 2u); }
+
+TEST_F(StorageTableTest, MoveConstructor) {
+  Table t2 = std::move(t);
+  EXPECT_EQ(t.chunk_count(), 0u);
```

Comment:
Accessing moved from objects might result in trouble. This should be covered during the next lecture.